src/api/routes/offer.js

```diff
@@ -98,14 +101,14 @@ export default (app) => {
         validators.offersDateSanitizers,
         async (req, res, next) => {
             try {
-
+                const account = await Account.findOne({ company: req.targetOwner });
+                const application = await CompanyApplication.findOne({ email: account.email });
```

Comment:
On way for this to "look cleaner" is if the "pending" state could be computed from the account object itself. Take a look at [Mongoose Schema Virtuals](https://mongoosejs.com/docs/tutorials/virtuals.html) and see if you agree with this.

Reply:
Also, maybe this could be hidden behind a service call? Don't we have an `AccountService` ?